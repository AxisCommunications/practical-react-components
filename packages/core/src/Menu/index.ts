export * from './BaseMenu'
export * from './Menu'
