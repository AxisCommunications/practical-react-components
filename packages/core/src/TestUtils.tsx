import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { PracticalProvider } from './Practical'

/*
  Only used for Jest Snapshot Testing
*/

export const TP = ({
  children,
}: {
  readonly children: React.ReactElement
}): JSX.Element => <PracticalProvider>{children}</PracticalProvider>

/**
 * Mock reference to practical-root
 */
const createNodeMock = (element: React.ReactElement) => {
  if (element.props.id !== 'practical-root') {
    return null
  }

  return {}
}

export const TestRender = (jsx: JSX.Element) => {
  return renderer.create(<TP>{jsx}</TP>, { createNodeMock }).toJSON()
}

export const TestText = (): JSX.Element => <span>Test</span>

/* eslint-disable-next-line react/display-name */
export const TestTextWithForwardRef = React.forwardRef<HTMLElement>(
  ({ ...props }, ref) => (
    <span {...props} ref={ref}>
      Test
    </span>
  )
)
