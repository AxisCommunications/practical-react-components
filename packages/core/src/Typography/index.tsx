import styled, { css } from 'styled-components'

export type TypographyVariant =
  | 'chip-tag-text'
  | 'explanatory-text'
  | 'caption'
  | 'navigation-group'
  | 'navigation-label'
  | 'card-title'
  | 'group-title'
  | 'column-heading'
  | 'header-title'
  | 'default-text'
  | 'dialog-heading'
  | 'page-heading'
  | 'button-text'
  | 'compact-label'

const TYPOGRAPHY_VARIANTS = {
  /**
   * Paragraph
   *
   * The sort of text that appears in blocks inside
   * e.g. a modal, should be easily readable.
   */
  'default-text': {
    as: 'p',
    css: css`
      font-size: ${({ theme }) => theme.font.size.regular};
      line-height: ${({ theme }) => theme.font.lineHeight.large};
    `,
  },

  /**
   * Descriptive
   *
   * The sort of text that elaborates on another
   * piece of information or context.
   * Should be small to be able to show a lot.
   */
  caption: {
    as: 'p',
    css: css`
      font-size: ${({ theme }) => theme.font.size.small};
      line-height: ${({ theme }) => theme.font.lineHeight.small};
    `,
  },

  /**
   * Dialog heading
   */
  'dialog-heading': {
    as: 'p',
    css: css`
      font-size: ${({ theme }) => theme.font.size.larger};
      line-height: ${({ theme }) => theme.font.lineHeight.larger};
    `,
  },

  /**
   * Page heading
   */
  'page-heading': {
    as: 'p',
    css: css`
      font-size: ${({ theme }) => theme.font.size.larger};
      line-height: ${({ theme }) => theme.font.lineHeight.larger};
    `,
  },

  /**
   * Chip or tag text
   *
   * Text appearing as chip or tag.
   * Should be small text.
   */
  'chip-tag-text': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.small};
      line-height: ${({ theme }) => theme.font.lineHeight.smaller};
      white-space: nowrap;
    `,
  },

  /**
   * Explanatory text
   *
   * Text appearing as explanatory text, e.g. error message in a field.
   * Should be small text.
   */
  'explanatory-text': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.small};
      line-height: ${({ theme }) => theme.font.lineHeight.small};
      white-space: nowrap;
    `,
  },

  /**
   * Navigation group
   *
   * Text appearing as navigation group name, e.g. Group name on sidebar.
   */
  'navigation-group': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.small};
      line-height: ${({ theme }) => theme.font.lineHeight.smaller};
      white-space: nowrap;
    `,
  },

  /**
   * Navigation label
   *
   * Standard size used for text labels,
   * short descriptions connected to another element.
   * e.g. Label on sidebar
   */
  'navigation-label': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.regular};
      line-height: ${({ theme }) => theme.font.lineHeight.small};
      white-space: nowrap;
    `,
  },

  /**
   * Card title
   */
  'card-title': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.regular};
      font-weight: ${({ theme }) => theme.font.fontWeight.semibold};
      line-height: ${({ theme }) => theme.font.lineHeight.small};
      white-space: nowrap;
    `,
  },

  /**
   * Group title
   *
   * Text appearing as a title of some fields or some group in Dialog.
   */
  'group-title': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.regular};
      font-weight: ${({ theme }) => theme.font.fontWeight.semibold};
      line-height: ${({ theme }) => theme.font.lineHeight.small};
      white-space: nowrap;
    `,
  },

  /**
   * Column heading in a table
   */
  'column-heading': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.small};
      font-weight: ${({ theme }) => theme.font.fontWeight.normal};
      line-height: ${({ theme }) => theme.font.lineHeight.smaller};
      white-space: nowrap;
    `,
  },

  /**
   * Header title
   */
  'header-title': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.regular};
      font-weight: ${({ theme }) => theme.font.fontWeight.semibold};
      line-height: ${({ theme }) => theme.font.lineHeight.small};
      white-space: nowrap;
    `,
  },

  /**
   * Button
   *
   * Text appearing inside a button
   */
  'button-text': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.large};
      font-weight: ${({ theme }) => theme.font.fontWeight.semibold};
      line-height: ${({ theme }) => theme.font.lineHeight.large};
      white-space: nowrap;
    `,
  },

  /**
   * Compact label
   *
   * Label where space is limited
   */
  'compact-label': {
    as: 'span',
    css: css`
      font-size: ${({ theme }) => theme.font.size.smaller};
      line-height: ${({ theme }) => theme.font.lineHeight.smallest};
    `,
  },
}

/**
 *
 * Generic typography
 *
 */

export interface TypographyProps {
  readonly variant?: TypographyVariant
}

export const Typography = styled.p.attrs<TypographyProps>(
  ({ variant = 'default-text' }) => ({
    as: TYPOGRAPHY_VARIANTS[variant].as,
  })
)<TypographyProps>`
  font-family: ${({ theme }) => theme.font.family};
  font-size: ${({ theme }) => theme.font.size.regular};
  line-height: ${({ theme }) => theme.font.lineHeight.large};
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0;

  ${({ variant = 'default-text' }) => TYPOGRAPHY_VARIANTS[variant].css}
`
