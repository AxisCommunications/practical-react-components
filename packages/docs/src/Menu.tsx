import React, { useCallback, useState, useMemo } from 'react'
import styled, { css } from 'styled-components'
import { useHistory, useLocation, matchPath } from 'react-router-dom'
import {
  ExpandableList,
  ExpandableListItemType,
  IconButton,
} from 'practical-react-components-core'
import { HamburgerMenuIcon } from 'practical-react-components-icons'

import { Components, Component } from './types'

const BASE_ITEMS = [
  {
    name: 'Home',
    route: '/',
  },
  {
    name: 'Changelog',
    route:
      'https://github.com/AxisCommunications/practical-react-components/blob/main/CHANGELOG.md',
  },
]

function groupBy<T, K>(list: ReadonlyArray<T>, getKey: (item: T) => K) {
  const map = new Map<K, Array<T>>()

  list.forEach(item => {
    const key = getKey(item)
    const collection = map.get(key)

    if (collection === undefined) {
      map.set(key, [item])
    } else {
      collection.push(item)
    }
  })

  return Array.from(map.values())
}

const MenuContainer = styled.div<{ readonly showNavigation: boolean }>`
  background-color: ${({ theme }) => theme.color.background00()};
  display: inline-flex;
  flex-direction: column;

  ${({ showNavigation }) =>
    showNavigation
      ? css`
          height: 100%;
          width: 300px;
        `
      : undefined}
`

const ExpandableListContainer = styled.div`
  overflow-y: auto;
`

interface MenuProps {
  readonly components: Components
}

interface ComponentGroup extends Component {
  readonly id: ExpandableListItemType['id']
  readonly label: ExpandableListItemType['label']
  readonly icon: ExpandableListItemType['icon']
  readonly selected: ExpandableListItemType['selected']
}

export const Menu: React.FC<MenuProps> = ({ components }) => {
  const [showNavigation, setShowNavigation] = useState<boolean>(true)
  const history = useHistory()
  const location = useLocation()
  const selectTab = useCallback((route: string) => history.push(route), [
    history,
  ])
  const onMenuIconClick = useCallback(
    () => setShowNavigation(!showNavigation),
    [showNavigation]
  )

  const mappedComponents = useMemo(
    () =>
      ([...BASE_ITEMS, ...components].map(c => ({
        id: c.name,
        label: c.name,
        icon: () => null,
        selected:
          matchPath(location.pathname, {
            path: c.route,
            exact: true,
          }) !== null,
        onClick: () =>
          /^https?:\/\//.test(c.route)
            ? (window.location.href = c.route)
            : selectTab(c.route),
        ...c,
      })) as unknown) as ReadonlyArray<ComponentGroup>,
    [location.pathname, components, selectTab]
  )

  const groupedComponents = useMemo(
    () =>
      groupBy(mappedComponents, item => item.menu).reduce(
        (acc: ReadonlyArray<ExpandableListItemType>, g) => {
          const nested = g[0].menu !== undefined
          const name: string = nested ? g[0].menu : ''

          if (!nested || name === '') {
            return [...acc, ...g]
          }

          return [
            ...acc,
            {
              id: name,
              label: name,
              icon: () => null,
              selected: history.location.pathname.startsWith(
                `/${name.toLocaleLowerCase()}`
              ),
              items: g,
            },
          ]
        },
        []
      ),
    [history.location.pathname, mappedComponents]
  )

  return (
    <MenuContainer showNavigation={showNavigation}>
      <IconButton
        variant="secondary"
        icon={HamburgerMenuIcon}
        onClick={onMenuIconClick}
      />

      {showNavigation ? (
        <ExpandableListContainer>
          <ExpandableList items={groupedComponents} />
        </ExpandableListContainer>
      ) : null}
    </MenuContainer>
  )
}
