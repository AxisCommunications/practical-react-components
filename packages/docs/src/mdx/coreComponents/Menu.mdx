export const meta = {
  name: 'Menu âœ”',
  route: '/components/menu',
  menu: 'Components',
}

import {
  MoreIcon,
  CheckIcon,
  ColumnsIcon,
} from 'practical-react-components-icons'

import { Menu } from 'practical-react-components-core'

# Menu

A Menu displays a list of choices on a temporary surface. It appears when the user interacts with a button, or other control.

## Demo

export const onClickHandler = label => {
  return () => alert(`clicked item: "${label}"`)
}

export const ITEMS = [
  { icon: CheckIcon, label: 'Enabled item' },
  { icon: ColumnsIcon, label: 'Column options', divider: true },
  { label: 'Item 3', disabled: true, danger: true },
]

<div
  style={{
    padding: '2px',
    display: 'inline-grid',
    gridGap: '10px',
    gridTemplateColumns: 'repeat(auto-fit, minmax(100px, 1fr))',
    width: '100%',
  }}
>
  <Menu
    items={ITEMS.map(({ label, ...item }) => ({
      ...item,
      label,
      onClick: onClickHandler(label),
    }))}
  />
  <Menu
    icon={MoreIcon}
    items={ITEMS.map(({ label, ...item }) => ({
      ...item,
      label,
      onClick: onClickHandler(label),
    }))}
  />
  <Menu
    items={ITEMS.map(({ label, ...item }) => ({
      ...item,
      label,
      onClick: onClickHandler(label),
    }))}
    disabled={true}
  />
  <Menu
    icon={MoreIcon}
    items={ITEMS.map(({ label, ...item }) => ({
      ...item,
      label,
      onClick: onClickHandler(label),
    }))}
    disabled={true}
  />
</div>

## Basic usage

```typescript type=live
<Menu
  icon={MoreIcon}
  items={[
    { icon: CheckIcon, label: 'Item 1', onClick: () => {} },
    { label: 'Item 2', onClick: () => {}, divider: true },
    { label: 'Item 3', onClick: () => {}, disabled: true, danger: true },
  ]}
/>
```

## Menu Props

Menu forwards all props to `PopOver`.
See [`PopOver props`](#components/popover)
for more details on which additional props that are available.

<Props of="Menu" />

## MenuWrapper Props

MenuWrapper forwards all unknown props to the base native `<div>` component. See
[`<div>` on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/div) for more
details on which additional props that are available.

<Props of="MenuWrapper" />
