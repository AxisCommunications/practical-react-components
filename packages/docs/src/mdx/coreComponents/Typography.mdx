export const meta = {
  name: 'Typography âœ”',
  route: '/styles/typography',
  menu: 'Styles',
}

import {
  SimpleTable,
  SimpleTableRow,
  Typography,
} from 'practical-react-components-core'

# Typography

## Design

Typography is used to present the design and content as clearly and efficiently as possible.
It contains reusable categories of text, each with an intended application and meaning.

<img
  src={require('../images/typographies.png').default}
  alt="Typography design"
/>

## Demo

export const TABLE_HEADER_DATA = [
  'Variant',
  'Font size',
  'Case with actual size',
]

export const TYPOGRAPHY_LIST = [
  { variant: 'page-heading', fontSize: '21px', case: 'Page heading' },
  { variant: 'dialog-heading', fontSize: '21px', case: 'Dialog heading' },
  {
    variant: 'button-text',
    fontSize: '14px',
    case: 'Regular button text',
  },
  {
    variant: 'default-text',
    fontSize: '13px',
    case: `Text in a paragraph. Lorem ipsum dolor sit amet, consectetur adipiscing elit,
  sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.`,
  },
  {
    variant: 'header-title',
    fontSize: '13px',
    case: 'Header title',
  },
  {
    variant: 'group-title',
    fontSize: '13px',
    case: 'Group title',
  },
  {
    variant: 'card-title',
    fontSize: '13px',
    case: 'Card title',
  },
  {
    variant: 'navigation-label',
    fontSize: '13px',
    case: 'A navigation label belong to something',
  },
  {
    variant: 'column-heading',
    fontSize: '12px',
    case: 'Use me for table headers',
  },
  {
    variant: 'navigation-group',
    fontSize: '12px',
    case: 'Navigation group',
  },
  {
    variant: 'caption',
    fontSize: '12px',
    case: 'Extra descriptive text used as a caption.',
  },
  {
    variant: 'explanatory-text',
    fontSize: '12px',
    case: 'A explanatory text, e.g. error message in a field. ',
  },
  {
    variant: 'chip-tag-text',
    fontSize: '12px',
    case: 'Piece of data in e.g. a chip or a tag',
  },
  {
    variant: 'compact-label',
    fontSize: '11px',
    case: 'Some label where space is limited',
  },
]

<SimpleTable
  maxHeight="400"
  header={TABLE_HEADER_DATA.map((cell, index) => (
    <Typography variant="column-heading" key={index}>
      {cell}
    </Typography>
  ))}
>
  {TYPOGRAPHY_LIST.map((typography, index) => (
    <SimpleTableRow key={index}>
      <Typography>{typography.variant}</Typography>
      <Typography>{typography.fontSize}</Typography>
      <Typography variant={typography.variant}>{typography.case}</Typography>
    </SimpleTableRow>
  ))}
</SimpleTable>

## Basic usage

```typescript type=live
<Typography variant="column-heading">Text</Typography>
```

## Props

Typography forwards all unknown props to the styled-components component. See
[`API Reference`](https://styled-components.com/docs/api) for more
details on which additional props that are available.

<Props of="Typography" />
