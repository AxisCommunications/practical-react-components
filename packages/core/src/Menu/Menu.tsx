import React, { memo, useMemo } from 'react'
import styled, { css, useTheme } from 'styled-components'

import { componentSize, spacing } from '../designparams'
import { Typography } from '../Typography'
import { Icon, IconType } from '../Icon'
import { BaseMenu, BaseItemProps, BaseMenuProps } from './BaseMenu'

export const MenuItem = styled.div<{
  readonly divider?: boolean
  readonly danger?: boolean
  readonly compact: boolean
}>`
  display: flex;
  align-items: center;
  padding: 0 ${spacing.large};
  > * {
    margin-right: ${spacing.large};
  }
  height: ${({ compact }) =>
    compact ? componentSize.small : componentSize.medium};

  border-bottom: ${({ divider, theme }) =>
    divider === true ? `1px solid ${theme.color.element12()}` : 0};

  ${({ danger, theme }) =>
    danger === true
      ? css`
          color: ${theme.color.textError()};
        `
      : undefined}
`

const MenuItemIcon = styled(Icon)`
  color: ${({ theme }) => theme.color.text05()};
`

export interface MenuItemProps
  extends Omit<BaseItemProps, 'component' | 'keyboardSelect'> {
  readonly icon?: IconType
  readonly label: string
  readonly divider?: boolean
  readonly danger?: boolean
  /**
   * Override theme's default setting for `compact` if set.

   */
  readonly compact?: boolean
}

interface MenuProps extends Omit<BaseMenuProps, 'components'> {
  /**
   * An array of items in the drop down menu.
   */
  readonly items: ReadonlyArray<MenuItemProps>
  /**
   * Override theme's default setting for `compact` if set.

   */
  readonly compact?: boolean
}

/**
 * Menu
 *
 * Forwards props to BaseMenu
 */
export const Menu = memo<MenuProps>(
  ({ items, compact: compactFromProps, ...props }) => {
    const { compact: compactFromTheme } = useTheme()
    const compact = compactFromProps ?? compactFromTheme

    /**
     * Creates array of components using MenuItem to
     * forward to Base Menu
     */
    const components = useMemo<ReadonlyArray<BaseItemProps>>(
      () =>
        items.map(item => {
          const { icon, label, divider, danger } = item

          const menuItemContent =
            icon !== undefined ? (
              <MenuItemIcon icon={icon} size="small" />
            ) : undefined

          return {
            component: (
              <MenuItem divider={divider} danger={danger} compact={compact}>
                {menuItemContent}
                <Typography variant="navigation-label">{label}</Typography>
              </MenuItem>
            ),
            ...item,
          }
        }),
      [compact, items]
    )

    return <BaseMenu components={components} {...props} />
  }
)

Menu.displayName = 'Menu'
