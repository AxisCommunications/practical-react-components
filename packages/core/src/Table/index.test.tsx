import React from 'react'
import 'jest-styled-components'

import { Table, TableHeader, TableRow, TableHeaderText } from '.'
import { TestRender } from '../TestUtils'
import { Typography } from '../Typography'
import { IMenuItem, Menu } from '../Menu'

const NOOP = () => {
  /** */
}
const clickHandler = () => console.log('Menu item click')

interface IDevice {
  readonly serialNumber: string
  readonly name: string
  readonly ip: string
  readonly disabled?: boolean
}

const TABLE_HEADER_DATA: ReadonlyArray<{
  readonly id: string
  readonly title: string
  readonly width: number
}> = [
  { id: 'serial-number', title: 'Serial number', width: 2 },
  { id: 'name', title: 'Name', width: 1 },
  { id: 'ip-address', title: 'IP address', width: 2 },
]

const DEVICE_LIST: ReadonlyArray<IDevice> = [
  { serialNumber: 'FACEFACE0001', name: 'a', ip: '192.168.0.1' },
  { serialNumber: 'FACEFACE0002', name: 'b', ip: '192.168.0.2' },
  {
    serialNumber: 'FACEFACE0003',
    name: 'c',
    ip: '192.168.0.3',
    disabled: true,
  },
  { serialNumber: 'FACEFACE0004', name: 'd', ip: '192.168.0.4' },
  { serialNumber: 'FACEFACE0005', name: 'e', ip: '192.168.0.5' },
  { serialNumber: 'FACEFACE0006', name: 'f', ip: '192.168.0.6' },
  { serialNumber: 'FACEFACE0007', name: 'g', ip: '192.168.0.7' },
  { serialNumber: 'FACEFACE0008', name: 'h', ip: '192.168.0.8' },
  { serialNumber: 'FACEFACE0009', name: 'i', ip: '192.168.0.9' },
  { serialNumber: 'FACEFACE0010', name: 'j', ip: '192.168.0.10' },
  { serialNumber: 'FACEFACE0011', name: 'k', ip: '192.168.0.11' },
  { serialNumber: 'FACEFACE0012', name: 'l', ip: '192.168.0.12' },
]

const ITEMS: ReadonlyArray<Omit<IMenuItem, 'onClick'>> = [
  { label: 'Item 1' },
  { label: 'Item 2', divider: true },
  { label: 'Item 3', disabled: true, danger: true },
]

describe('Tables', () => {
  test('Table (minimal)', () => {
    const tree = TestRender(
      <Table maxHeight={6}>
        <TableHeader>
          {TABLE_HEADER_DATA.map(({ title }, id) => (
            <TableHeaderText key={id}>{title}</TableHeaderText>
          ))}
        </TableHeader>
        {DEVICE_LIST.map(device => (
          <TableRow
            id={device.serialNumber}
            key={device.serialNumber}
            disabled={device.disabled}
          >
            <Typography>{device.serialNumber}</Typography>
            <Typography>{device.name}</Typography>
            <Typography>{device.ip}</Typography>
          </TableRow>
        ))}
      </Table>
    )
    expect(tree).toMatchSnapshot()
  })
  test('Table (full)', () => {
    const tree = TestRender(
      <Table
        initialWidths={TABLE_HEADER_DATA.map(({ width }) => width)}
        maxHeight={6}
        onSelect={NOOP}
        hasMenu={true}
      >
        <TableHeader
          selected={true}
          partial={false}
          overlay={
            <div>
              <Typography>Actions overlay</Typography>
            </div>
          }
          menu={
            <Menu
              align="right"
              items={ITEMS.map(({ label, ...item }) => ({
                ...item,
                label,
                onClick: clickHandler,
              }))}
            />
          }
        >
          {TABLE_HEADER_DATA.map(({ title }, id) => (
            <TableHeaderText key={id}>{title}</TableHeaderText>
          ))}
        </TableHeader>
        {DEVICE_LIST.map(device => (
          <TableRow
            id={device.serialNumber}
            key={device.serialNumber}
            selected={true}
            disabled={device.disabled}
            menu={
              <Menu
                align="right"
                items={ITEMS.map(({ label, ...item }) => ({
                  ...item,
                  label,
                  onClick: clickHandler,
                }))}
              />
            }
          >
            <Typography>{device.serialNumber}</Typography>
            <Typography>{device.name}</Typography>
            <Typography>{device.ip}</Typography>
          </TableRow>
        ))}
      </Table>
    )
    expect(tree).toMatchSnapshot()
  })
})
