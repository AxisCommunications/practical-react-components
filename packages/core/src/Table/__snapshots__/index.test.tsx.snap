// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tables Table (full) 1`] = `
.c2 {
  overflow-x: auto;
  overflow-y: hidden;
  position: relative;
  height: 100%;
  width: 100%;
  background-color: rgb(255,255,255);
}

.c3 {
  display: grid;
  width: 100%;
  border-bottom: 1px solid rgb(230,230,230);
}

.c16 {
  display: grid;
  overflow-y: auto;
  overflow-x: hidden;
  padding-bottom: 8px;
  position: relative;
  z-index: 0;
}

.c15 {
  font-family: Open Sans,Helvetica,arial,sans-serif;
  font-size: 13px;
  line-height: 22px;
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0;
  font-size: 13px;
  line-height: 22px;
}

.c10 {
  position: absolute;
  top: 3px;
  left: 3px;
  height: 18px;
  width: 18px;
  border-radius: 2px;
  -webkit-transition: -webkit-transform 100ms;
  -webkit-transition: transform 100ms;
  transition: transform 100ms;
  pointer-events: none;
}

.c8 {
  position: relative;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 24px;
  height: 24px;
}

.c8:hover .c9 {
  background-color: rgba(103,58,183,0.16);
  -webkit-transform: scale(1.77);
  -ms-transform: scale(1.77);
  transform: scale(1.77);
}

.c8:focus-within {
  outline: none;
}

.c8:focus-within .c9 {
  background-color: rgba(103,58,183,0.16);
  -webkit-transform: scale(1.77);
  -ms-transform: scale(1.77);
  transform: scale(1.77);
}

.c8:active .c9 {
  background-color: rgba(103,58,183,0.24);
  -webkit-transform: scale(1.88);
  -ms-transform: scale(1.88);
  transform: scale(1.88);
}

.c13 {
  position: absolute;
  top: 0;
  left: 0;
  cursor: pointer;
  display: unset;
  margin: 0;
  padding: 0;
  height: 100%;
  width: 100%;
  opacity: 0;
}

.c11 {
  width: 18px;
  height: 18px;
  border-radius: 2px;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  position: absolute;
}

.c12 {
  background-color: rgb(103,58,183);
}

.c12::after {
  content: '';
  width: 5px;
  height: 10px;
  border: solid #fff;
  border-width: 0 2px 2px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  margin-bottom: 3px;
}

.c5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  box-sizing: border-box;
  height: 100%;
  width: 100%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c6 {
  color: rgb(61,61,61);
  background: rgb(255,255,255);
  text-align: left;
  font-weight: 400;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c6:not:last-child {
  resize: horizontal;
}

.c7 {
  padding-right: 24px;
  grid-column: start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: visible;
}

.c14 {
  z-index: 1;
  grid-column-start: content-start;
  grid-column-end: end;
  color: rgb(69,39,160);
}

.c4 {
  display: grid;
  grid-template-rows: 48px;
}

.c18 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c17 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-3"
    >
      <div
        className="c2"
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            style={
              Object {
                "gridTemplateColumns": "24px [start] 48px [content-start] 2px 1px 2px [content-end] 40px [scroll] 12px [end]",
              }
            }
          >
            <div
              className="c5 c6 c7"
            >
              <div
                checked={true}
                className="c8"
                disabled={false}
              >
                <div
                  className="c9 c10"
                />
                <div
                  className="c11 c12"
                />
                <input
                  checked={true}
                  className="c13"
                  disabled={false}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onPointerDown={[Function]}
                  onPointerUp={[Function]}
                  type="checkbox"
                />
              </div>
            </div>
            <div
              className="c5 c14"
            >
              <div>
                <p
                  className="c15"
                >
                  Actions overlay
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="c16"
          style={
            Object {
              "maxHeight": "336px",
              "width": "129px",
            }
          }
          width={129}
        />
      </div>
    </div>
  </div>
  <div>
    <div
      className="c17"
      id="layer-4"
    >
      <div
        className="c18"
      />
    </div>
  </div>
</div>
`;

exports[`Tables Table (minimal) 1`] = `
.c2 {
  overflow-x: auto;
  overflow-y: hidden;
  position: relative;
  height: 100%;
  width: 100%;
  background-color: rgb(255,255,255);
}

.c3 {
  display: grid;
  width: 100%;
  border-bottom: 1px solid rgb(230,230,230);
}

.c10 {
  display: grid;
  overflow-y: auto;
  overflow-x: hidden;
  padding-bottom: 8px;
  position: relative;
  z-index: 0;
}

.c8 {
  font-family: Open Sans,Helvetica,arial,sans-serif;
  font-size: 13px;
  line-height: 22px;
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0;
  font-size: 12px;
  font-weight: 400;
  line-height: 17px;
  white-space: nowrap;
}

.c5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  box-sizing: border-box;
  height: 100%;
  width: 100%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c6 {
  color: rgb(61,61,61);
  background: rgb(255,255,255);
  text-align: left;
  font-weight: 400;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c6:not:last-child {
  resize: horizontal;
}

.c7 {
  padding-right: 16px;
}

.c7:first-child {
  grid-column: content-start;
}

.c9 {
  text-transform: none;
}

.c4 {
  display: grid;
  grid-template-rows: 48px;
}

.c12 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c11 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-1"
    >
      <div
        className="c2"
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            style={
              Object {
                "gridTemplateColumns": "24px [start] 0px [content-start] 64px 64px 64px [content-end] 0px [scroll] 12px [end]",
              }
            }
          >
            <div
              className="c5 c6 c7"
            >
              <p
                className="c8 c9"
              >
                Serial number
              </p>
            </div>
            <div
              className="c5 c6 c7"
            >
              <p
                className="c8 c9"
              >
                Name
              </p>
            </div>
            <div
              className="c5 c6 c7"
            >
              <p
                className="c8 c9"
              >
                IP address
              </p>
            </div>
          </div>
        </div>
        <div
          className="c10"
          style={
            Object {
              "maxHeight": "336px",
              "width": "228px",
            }
          }
          width={228}
        />
      </div>
    </div>
  </div>
  <div>
    <div
      className="c11"
      id="layer-2"
    >
      <div
        className="c12"
      />
    </div>
  </div>
</div>
`;
